src/api/user/model/model.ts

```diff
@@ -0,0 +1,54 @@
+import { v4 as uuidv4 } from 'uuid';
+
+import { CreateUserParams, UpdateUserParams, User } from './types';
+import { defaultState } from './_defaultState';
+
+export class UserModel {
```

Comment:
it looks like the Repository but not the Model level. Model means one entity but here you handle the entities.